### PR TITLE
Revendor `gardener/gardener@v1.40.0`

### PR DESCRIPTION
**Pull request by dkistner**
**How to categorize this PR?**
/kind enhancement
/platform aws
/merge squash

**What this PR does / why we need it**:
Revendor `gardener/gardener@v1.40.0` dependency.
This will include an upgrade to `ginkgo` v2.

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by dkistner** [comment]
/invite @timebertt regarding the ginkgo v2 upgrade 
/invite @kon-angelo

File: test/integration/bastion/bastion_test.go [hunk 1]
```diff
@@ -84,107 +84,111 @@ func validateFlags() {
...
-			mgrCancel()
-		}()
+	amiID := determineBastionImage(ctx, awsClient)
+	extensionscluster, corecluster = newCluster(namespaceName, amiID)
+})
 
-		By("running cleanup actions")
-		framework.RunCleanupActions()
+var _ = AfterSuite(func() {
+	defer func() {
```
Comment (timebertt) [inline test/integration/bastion/bastion_test.go:179]: You could use `DeferCleanup` https://onsi.github.io/ginkgo/MIGRATING_TO_V2#defercleanup

**Review by timebertt — DISMISSED** [review]
/lgtm

File: test/integration/bastion/bastion_test.go (hunk 1 above)
Comment (dkistner) [reply test/integration/bastion/bastion_test.go]: done with `5dbbbfc`

**Review by timebertt — APPROVED** [review]
/lgtm

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Pull request merged**